OutOfSchool/OutOfSchool.BusinessLogic/Services/ContactsService.cs

```diff
@@ -0,0 +1,113 @@
+using System.Diagnostics.CodeAnalysis;
+using AutoMapper;
+using OutOfSchool.BusinessLogic.Models;
+using OutOfSchool.BusinessLogic.Models.ContactInfo;
+using OutOfSchool.Services.Models.ContactInfo;
+
+namespace OutOfSchool.BusinessLogic.Services;
+
+public class ContactsService<TEntity, TDto> : IContactsService<TEntity, TDto>
+    where TEntity : BusinessEntity, IHasContacts
+    where TDto : IHasContactsDto<TEntity>
+{
+    private readonly IMapper mapper;
+
+    public ContactsService(IMapper mapper)
```

Comment:
Please consider usage of primary constructor. 
There is one downside of this approach (you can't `throw ArgumentNullException`), but for DI injected params it's not a big issue, I guess